ckan/model/activity.py

```diff
@@ -136,7 +141,21 @@ def _activities_limit(q: QActivity, limit: int,
     q = q.order_by(desc(model.Activity.timestamp))  # type: ignore
     if offset:
         q = q.offset(offset)
+
+    if after:
+        q = q.filter(model.Activity.timestamp > after)
+    elif before:
+        q = q.filter(model.Activity.timestamp < before)
+
     if limit:
+        # if we want activities after some timestamp, we need to limit
+        # last N rows from this query
+        if after:
+            total = q.count()
+            offset = total - limit
+            # going back until the first page, we don't need to offset
+            if offset > 0:
+                q = q.offset(offset)
```

Comment:
instead of getting all results when using then applying an offset, use an ascending `order_by` and then reverse the results returned.

Reply:
We now [reverse the order](https://github.com/avdata99/ckan/blob/6108_improve_pkg_actvity_list/ckan/model/activity.py#L143-L147) when we [look for `before` without `after`](https://github.com/avdata99/ckan/blob/6108_improve_pkg_actvity_list/ckan/model/activity.py#L266-L272)